src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/CommonCryptographyManager.cs

```diff
@@ -53,15 +53,11 @@ public byte[] CreateSha256HashBytes(string input)
             }
         }
 
-        /// <inheritdoc />
+        /// <remarks>AAD only supports RSA certs for client credentials </remarks>
         public virtual byte[] SignWithCertificate(string message, X509Certificate2 certificate)
         {
-            if (certificate.GetRSAPublicKey().KeySize < CertificateClientCredential.MinKeySizeInBits)
-            {
-                throw new ArgumentOutOfRangeException(nameof(certificate),
-                    string.Format(CultureInfo.InvariantCulture, MsalErrorMessage.CertificateKeySizeTooSmallTemplate,
-                        CertificateClientCredential.MinKeySizeInBits));
-            }
+            // MSAL used to check min key size by looking at certificate.GetRSAPublicKey().KeySize
+            // but this causes sporadic failures in the crypto stack. Rely on AAD to perform key size validations.
```

Comment:
Do you know why we validated this if it was also validated by AAD?

Reply:
This is very old code, copied from ADAL, so no history on this. 
I am not saying that AAD validates cert size, I am saying that it's AAD's responsability to validate, since the cert needs to be uploaded there.